src/webgpu/api/validation/createSampler.spec.ts

```diff
@@ -24,3 +24,23 @@ g.test('lodMinAndMaxClamp')
       });
     }, t.params.lodMinClamp > t.params.lodMaxClamp || t.params.lodMinClamp < 0 || t.params.lodMaxClamp < 0);
   });
+
+g.test('maxAnisotropy')
+  .desc('test different maxAnisotropy values and combinations with min/mag/mipmapFilter')
+  .params(
+    params()
+      .combine(poptions('maxAnisotropy', [0, 1, 2, 4, 16, 32, 1024]))
+      .combine(poptions('minFilter', ['nearest', 'linear']))
+      .combine(poptions('magFilter', ['nearest', 'linear']))
+      .combine(poptions('mipmapFilter', ['nearest', 'linear']))
+  )
+  .fn(async t => {
+    t.expectValidationError(() => {
+      t.device.createSampler({
+        minFilter: t.params.minFilter as GPUFilterMode,
+        magFilter: t.params.magFilter as GPUFilterMode,
+        mipmapFilter: t.params.mipmapFilter as GPUFilterMode,
+        maxAnisotropy: t.params.maxAnisotropy,
+      });
```

Comment:
Since this is a validation test it's not really necessary to hit this entire test space - most of them are invalid for multiple reasons so they don't add coverage.

It's possible to reduce the combinatorics with something like this:
```suggestion
  .params([
    ...poptions('maxAnisotropy', [-1, undefined, 0, 1, 2, 4, 7, 16, 32, 33, 1024]),
    { minFilter: 'nearest' as const },
    { magFilter: 'nearest' as const },
    { mipmapFilter: 'nearest' as const },
  ])
  .fn(async t => {
    const {
      maxAnisotropy = 1,
      minFilter = 'linear',
      magFilter = 'linear',
      mipmapFilter = 'linear',
    } = t.params;
    t.expectValidationError(() => {
      t.device.createSampler({ minFilter, magFilter, mipmapFilter, maxAnisotropy });
```

Reply:
Having this error when trying to add null, NaN to the test case. Is typescript compiler making these the same type as undefined?
`Duplicate public test case params for test maxAnisotropy: maxAnisotropy=null`

Modified a bit to pass code style check, might be suboptimal.